packages/labs/ssr-dom-shim/src/lib/css-loader.ts

```diff
@@ -0,0 +1,42 @@
+import {readFileSync} from 'node:fs';
+import type {LoadHook, ResolveHook} from 'node:module';
+import {fileURLToPath} from 'node:url';
+
+/**
+ * Checks for each import, whether the file exists and if not, tries
+ * to find an associated TypeScript file.
+ *
+ * https://nodejs.org/api/module.html#resolvespecifier-context-nextresolve
+ */
+export const resolve: ResolveHook = (specifier, context, nextResolve) => {
+  if (context.importAttributes.type === 'css') {
+    return {
+      format: 'module',
+      shortCircuit: true,
+      url: new URL(specifier, context.parentURL).toString(),
+    };
+  }
+  return nextResolve(specifier, context);
+};
+
+/**
+ * When an attempt is made to import a CSS file/module, the CSS content
+ * is read and added to a CSSRule, which is provided in a CSSStyleSheet
+ * instance.
+ *
+ * https://nodejs.org/api/module.html#loadurl-context-nextload
+ */
+export const load: LoadHook = (url, context, nextLoad) => {
+  if (context.importAttributes.type === 'css') {
```

Comment:
Could we add a warning if there's a `.css` import without the correct import attribute? And can loaders compose so that we only warn if that import otherwise failed?

Also, what happens if Node gains support for CSSStyleSheet natively and this loader is registered? Is there a way to detect that case?

Reply:
Yes, but only in the load hook. I'll implement a suggestion.

I am unaware of a possibility to detect what import attributes are natively supported. However, before registering our loader we could try a dynamic import of a css file and only register it, if that fails. That seems a bit hacky to me, but I am open to do that, if you would prefer that approach.